Components/AppilcationUpdate/Plugins/ApplicationUpdatePlugin.cs

```diff
@@ -0,0 +1,166 @@
+﻿using Serilog;
+using Slipstream.Shared;
+using Slipstream.Shared.Helpers.StrongParameters;
+using Slipstream.Shared.Helpers.StrongParameters.Validators;
+using Squirrel;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using Slipstream.Components.Internal.Events;
+
+namespace Slipstream.Components.AppilcationUpdate.Plugins
+{
+    public class ApplicationUpdatePlugin : BasePlugin
+    {
+        private readonly EventHandler.ApplicationUpdateEventHandler applicationUpdate;
+        private readonly IApplicationUpdateEventFactory applicationUpdateEventFactory;
+        private readonly IEventBus eventBus;
+        private string updateLocation;
+        private bool prerelease;
+        private bool updateNotified = false;
+
+        public ILogger Logger { get; }
+        internal static DictionaryValidator ConfigurationValidator { get; }
+
+        static ApplicationUpdatePlugin()
+        {
+            ConfigurationValidator = new DictionaryValidator()
+                .RequireString("updateLocation")
+                .PermitBool("prerelease");
+        }
+
+        public ApplicationUpdatePlugin(
+            IEventHandlerController eventHandlerController, 
+            string id, 
+            IApplicationUpdateEventFactory applicationUpdateEventFactory, 
+            ILogger logger, 
+            IEventBus eventBus, 
+            Parameters configuration)
+            : base(eventHandlerController, id, nameof(ApplicationUpdatePlugin), id, true)
+        {
+            ConfigurationValidator.Validate(configuration);
+
+            this.applicationUpdateEventFactory = applicationUpdateEventFactory;
+            Logger = logger;
+            this.eventBus = eventBus;
+
+            SubscribeToInternalEvents();
+
+            if (Debugger.IsAttached)
+            {
+                Logger.Information("Auto update is disabled when a Debugger is attached");
+                return;
+            }
+
+            //var updateLocation = configuration.ExtractOrDefault("updateLocation", string.Empty); 
+            //var prerelease = configuration.ExtractOrDefault("prerelease", false);
```

Comment:
You can remove these - they are a few lines down as well

Reply:
Removed